Library/Homebrew/cask/lib/hbc/installer.rb

```diff
@@ -172,6 +175,17 @@ def extract_primary_container
       else
         primary_container.extract_nestedly(to: @cask.staged_path, basename: basename, verbose: verbose?)
       end
+
+      return unless quarantine? && Quarantine.available?
+
+      unless Quarantine.detect(@downloaded_path)
+        raise CaskError, "#{@downloaded_path} was not quarantined properly."
+      end
+
+      @cask.staged_path.each_child do |path|
```

Comment:
Why not just call `Quarantine.all` on `@cask.staged_path`?

Reply:
Unpacking non-DMG containers (`.tar.gz` and other Unix-based stuff) does not propagate the container's extended attributes; instead, the *contents* of `@cask.staged_path` themselves have to be quarantined after extraction.

Reply:
(I'll simplify and remove this logic to `Quarantine.all`.)